### PR TITLE
[MM-30978] avoid parsing nil PostAction in SlackAttachments

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Avoid parsing nil/null `PostAction` in Posts.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-30978

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — CHANGES_REQUESTED** [review]
Small improvment request.

File: model/post.go [hunk 1]
```diff
@@ -557,6 +557,13 @@ func (o *Post) Attachments() []*SlackAttachment {
 			if enc, err := json.Marshal(attachment); err == nil {
 				var decoded SlackAttachment
 				if json.Unmarshal(enc, &decoded) == nil {
+					for i := 0; i < len(decoded.Actions); i++ {
+						if decoded.Actions[i] == nil {
+							copy(decoded.Actions[i:], decoded.Actions[i+1:])
+							decoded.Actions = decoded.Actions[:len(decoded.Actions)-1]
+							i--
+						}
+					}
```
[the remark this answers is not in the document]
Comment (isacikgoz) [reply model/post.go]: wow amazing, thanks :)

File: model/post_test.go [hunk 2]
```diff
@@ -884,3 +884,44 @@ func TestSearchParameterFromJson(t *testing.T) {
...
+
+	t.Run("a couple of actions", func(t *testing.T) {
+		p.Props["attachments"] = []interface{}{
+			map[string]interface{}{"actions": []interface{}{
+				map[string]interface{}{"id": "test1"}, map[string]interface{}{"id": "test2"}},
+			},
+		}
+
+		attachments := p.Attachments()
+		require.Len(t, attachments[0].Actions, 2)
```
Comment (streamer45) [inline model/post_test.go:914]: I understand the filtering is rather straightforward but would it make sense to also check the expected values, just to be extra confident? 

Same would apply for the case below.

Comment (isacikgoz) [reply model/post_test.go:914]: Fair enough 👍

**Review by agnivade — APPROVED** [review]
Looks great

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

**Comment by isacikgoz** [comment]
@lindalumitchell Since this is an edge case, probably no additional testing required apart from unit tests. But it can be tested with the test data attached in the ticket if it is necessary.

**Review by lindalumitchell — APPROVED** [review]
Thanks; I think it's fine.

**Pull request merged**